inference-engine/tests/functional/shared_test_classes/src/subgraph/simple_if.cpp

```diff
@@ -4,36 +4,43 @@
 
 #include "shared_test_classes/subgraph/simple_if.hpp"
 #include "ngraph_functions/builders.hpp"
+#include "functional_test_utils/ov_tensor_utils.hpp"
 
 namespace SubgraphTestsDefinitions {
 std::string SimpleIfTest::getTestCaseName(const testing::TestParamInfo<SimpleIfParamsTuple> &obj) {
-    std::vector<std::vector<size_t>> inputShapes;
-    InferenceEngine::Precision netPrecision;
+    std::vector<ov::test::InputShape> shapes;
```

Comment:
shouldn't we modify test to check non const `condition` value?
what about `then` and `else` branch with internal dynamism?

Reply:
There were 3 tests: 2 with constant condition and non constant. The first and second tests used transformation to fold If because we always know which branch we are going to use. The third test with non constant I have already updated

Moreover I added the fourth test with non constant condition and different output shapes: one branch has `eltwise`, another branch has `pad`. So we don't know before inference what output shapes are going to be. Do you think that it isn't enough? And should we add one more test with exact internal dynamism?

Reply:
it's not blocker for merge, but I guess we need to add such type test cases in future in `if` and `TI`